### DISCUSSION TITLE
next.js app deployment errror

### DISCUSSION THREAD

**Discussion by AhmadNasser04** [vercel/next.js#48248] [Help] [answered]
### Summary

I changed a bunch of things in my app, but the most noteworthy is that I added user authentication using Firebase, which is what I assume caused the issue.
I'm also using GraphQL for blogs, had an issue with that before, but resolved it.
Nothing else should cause this issue should it?

This is my github repo: https://github.com/AhmadNasser04/LBS
This is the link to the currently deployed version: https://lbs-one.vercel.app/
Image of Error: https://gyazo.com/6d722499dbfd57affbf5c9b255cb1102

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This error happens if the first argument passed to `request` is undefined. Make sure your:

```js
const graphqlAPI = process.env.NEXT_PUBLIC_GRAPHCMS_ENDPOINT;
```

Is properly defined in the Vercel dashboard for your project.

**Reply by AhmadNasser04** [reply]
Do I need to make it into an environment variable? I thought next did that on its own

**Reply by icyJoseph** [reply]
It does, but where does it read it from? Either . env files or from the Vercel Dashboard where you can fill-in various key-value pairs. 

We could confirm that the issue is that this value is missing by logging it inside getPost, and see what the build logs output.

**Reply by AhmadNasser04** [reply]
That was the issue, fixed, thankyou very much.
